src/KeyFinder.ts

```diff
@@ -116,32 +121,41 @@ export default class KeyFinder {
      * @throws {Error} Thrown if the AnswerHubAPI call cannot be completed or results in a status code other than 200 or 403
      */
     private async testKey(key: string): Promise<string | null> {
-        const resp = await fetch("https://euw1.api.riotgames.com/lol/summoner/v3/summoners/22929336", {
-            headers: {
-                "X-Riot-Token": key,
-            },
-        });
+        let resp;
+        try {
+            resp = await fetch("https://euw1.api.riotgames.com/lol/summoner/v3/summoners/22929336", {
+                headers: {
+                    "X-Riot-Token": key,
+                },
+            });
+        } catch (error) {
+            console.error(`Error occurred while making a request to the riot games api: ${error}`);
+            return null;
+        }
 
         return resp.status === 403 ? null : resp.headers.get("x-app-rate-limit");
     }
 
     /**
      * Tests all keys to see if they are still active, removing deactivated keys from the list and logging a message for each one
      */
-    private testAllKeys(): void {
+    private async testAllKeys() {
         for (let i = 0; i < this.keys.length; i++) {
             const keyInfo = this.keys[i];
-            this.testKey(keyInfo.apiKey).then(header => {
+
+            try {
+                const header = await this.testKey(keyInfo.apiKey);
                 if (header !== null) {
                     return;
                 }
+            } catch (error) {
+                console.error(`Error occurred while making a request to the riot games api: ${error}`);
```

Comment:
If I'm reading this correctly it will remove keys that error from the database, even though they might still be valid. Is that intentional?

Reply:
Also, just like the other place where this is called, `testKey` will never reject so this will never get called.